openedx/core/djangoapps/schedules/management/commands/set_up_models_for_sending_test_emails.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+import pytz
+
+from django.core.management.base import BaseCommand
+from student.models import CourseEnrollment
+from django.contrib.sites.models import Site
+from openedx.core.djangoapps.schedules.models import Schedule, ScheduleConfig
+
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        self._create_schedule_config_if_none_exists()
+        self._create_schedule_if_none_exist()
+        self._update_schedules()
+
+    def _create_schedule_config_if_none_exists(self):
+        schedule_configs = self._get_all_schedule_configs()
+        if len(schedule_configs) == 0:
+            self._create_schedule_config()
+            print "Creating ScheduleConfig."
+
+    def _create_schedule_if_none_exist(self):
+        schedules = self._get_all_schedules()
+        if len(schedules) == 0:
+            a_schedule = self._create_schedule()
+            print "Creating schedule.\nstart: {}\nupgrade_deadline: {}".format(a_schedule.start,
+                                                                               a_schedule.upgrade_deadline)
+
+    def _update_schedules(self):
+        schedules = self._get_all_schedules()
+        assert(len(schedules) > 0)
+
+        for a_schedule in schedules:
+            # TODO: Rewrite _calculate_n_days_from_now to be _calculate_n_days_from_date_m. Rewrite all offset calcs in
+            #       terms of new function
+            three_days_after_start = a_schedule.start + datetime.timedelta(days=3)
+            twenty_one_days_before_upgrade_deadline = a_schedule.upgrade_deadline - datetime.timedelta(days=21)
+
+            if self._today_is_three_days_after_start(three_days_after_start):
+                new_start = self._calculate_n_days_from_now(n=-3)
+                print "Updating 'start' from {0} to {1}".format(a_schedule.start, new_start)
+                a_schedule.start = new_start
+
+            if self._today_is_twenty_one_days_before_upgrade_deadline(twenty_one_days_before_upgrade_deadline):
+                new_upgrade_deadline = self._calculate_n_days_from_now(n=21)
+                print "Updating 'upgrade_deadline' from {0} to {1}".format(a_schedule.upgrade_deadline,
+                                                                           new_upgrade_deadline)
+                a_schedule.upgrade_deadline = new_upgrade_deadline
+
+            a_schedule.save()
+
+    def _get_all_schedule_configs(self):
+        return ScheduleConfig.objects.all()
+
+    def _get_all_schedules(self):
+        return Schedule.objects.all()
+
+    def _create_schedule(self):
+        upgrade_deadline = self._calculate_n_days_from_now(n=21)
+        honor_enrollment = CourseEnrollment.objects.get(id=1)
+        honor_enrollment.dynamic_upgrade_deadline = upgrade_deadline
```

Comment:
I'm hoping this doesn't work? `dynamic_upgrade_deadline` should be a read-only property